1.Use code generator replace codedom and upgrate to .net6

leeveel opened the pull request:
1.use code generator replace codedom
2.upgrate to .net6

leeveel committed:
1.CodeDom is replace by CodeGenerator
2.upgrate to .net6

jbakic commented:
Hi @leeveel,

Thank you for this contribution! I think this is probably the best way to upgrade the old proxy factory.

I have some comments on the changes, and I have noticed some issues. Would you maybe have the time to look at these? If not, just let me know, and I will accept the pull request and make the changes myself after the merge.

Below I will make a note of everything I noticed, including minor stuff that I can easily change myself. Feel free to do only some of these and skip some, anything you like.

What I noticed:
* The new project should probably be named Shielded.SourceGenerator.
* This one is for me for later, maybe - I'd like to get rid of the reference to Scriban. I generally avoid adding NuGet refs, to keep it as small as possible and avoid any issues for library users. (Though I'm not sure if a source generator can even cause diamond dependency issues?..)
* Comments in Chinese (I think) in a couple of places. ShieldedPocoGenerator line 33, and ConsoleTests.csproj line 15. Please translate to English or delete them.
* Missing XML documentation comments on the new attribute and interfaces that were added to IShielded.cs. I can add those if you like.
* The code generator could be improved to not insist on equal accessibility of getter and setter of properties, maybe? This was like that only because of a CodeDOM limitation...
* ResLoader.cs line 13 looks like it can be deleted.
* Factory.NewShielded should probably throw an exception if it cannot provide a result, like the exception in line 44.
* Factory line 44, mistake in the message - should say "[Shielded]", not "[Shield]".
* But the whole PrepareTypes method should probably just be deleted? It's not needed anymore.
* CommuteTests, the big comment at the beginning can be removed.
* A bigger one: Some unit tests are red. I added the source generator reference, and the attribute to the types where needed, but then this uncovered a bug in the source generator. It does not handle well the property named "switch" (ProxyTestEntities.cs line 22). It generates code that has `@@switch` in one place, which is a syntax error.
* Also, you can avoid some of the `@`s in the generated code, no C# keyword starts with `__struct`.
* The BetShop.cs class also has some classes that need the new attribute. These work great once that is added. (Uncomment the BetShopTest method in the main method of Program.cs to test this.)

As I said, I can take care of these things myself. After it is merged and I've tested it some more, I will create a 5.0.0 release and push it to NuGet. (I might also add/remove some features, which I've been thinking of for some time, before going for 5.0.0.) I will also change the LICENCE (it mentions only me in the copyright notice) and README files.

Thanks again for the contribution!

Josip

leeveel commented:
Hi @jbakic 
Thank you for your detailed review, I will fix all issues you listed later.

leeveel committed:
1.rename to Shielded.SourceGenerator
2.translate Chinese comments to English
3.support non-insist modifier for getter and setter
4.remove unless code in ResLoader.cs
5.throw exception when error in Factory.NewShielded method
6.fix mistake message tip in Factory.NewShielded method ("[Shield]" ==> "[Shielded]")
7.add a warm up for Factory.PrepareTypes
8.remove CommuteTests useless comment
9.fix keyword syntax error in code generator (such as:@switch)
10.add [shielded] attribute for pocos in BetShop.cs
11.commented ProtectedSetterTest

leeveel commented:
Hi @jbakic 
I have fixed all issues you listed except Srciban dependency.

1.rename to Shielded.SourceGenerator
2.translate Chinese comments to English
3.support non-insist modifier for getter and setter
4.remove useless code in ResLoader.cs
5.throw exception when error in Factory.NewShielded method
6.fix mistake message tip in Factory.NewShielded method ("[Shield]" ==> "[Shielded]")
7.add a warm up for Factory.PrepareTypes
8.remove CommuteTests useless comment
9.fix keyword syntax error in code generator (such as:@switch) 
10.add [shielded] attribute for pocos in BetShop.cs
11.commented ProtectedSetterTest, maybe this is no need anymore due to [3]???
12.pass all test unit

best regards!
leeveel

leeveel committed:
add test for non insist getter setter
